test/trento/infrastructure/commanded/event_handlers/database_deregistration_event_handler_test.exs

```diff
@@ -0,0 +1,31 @@
+defmodule Trento.Infrastructure.Commanded.EventHandlers.DatabaseDeregistrationEventHandlerTest do
+  use ExUnit.Case
+  use Trento.EventStoreCase
+
+  import Mox
+  import Trento.Factory
+
+  alias Trento.Databases.Events.DatabaseDeregistered
+  alias Trento.Infrastructure.Commanded.EventHandlers.DatabaseDeregistrationEventHandler
+  alias Trento.SapSystems.Commands.DeregisterSapSystem
+
+  setup [:set_mox_from_context, :verify_on_exit!]
+
+  test "should dispatch DeregisterSapSystem commands when a database is deregistered" do
+    %{id: database_id} = insert(:database)
+    %{id: sap_system_id} = insert(:sap_system, database_id: database_id)
```

Comment:
It would be nice to add a list of sap systems. To see that the command is sent for both

Reply:
done